api/v1beta1/remotesecret_types.go

```diff
@@ -111,21 +112,50 @@ type RemoteSecret struct {
 }
 
 var secretTypeMismatchError = errors.New("the type of upload secret and remote secret spec do not match")
+var secretDataKeysMissingError = errors.New("the secret data does not contain the required keys")
 
-// ValidateUploadSecretType checks weather the uploadSecret type matches the RemoteSecret type.
+// ValidateUploadSecret checks weather the uploadSecret type matches the RemoteSecret type and whether upload secret
+// contains required keys.
 // The function is in the api package because it extends the contract of the CRD.
-// In the future the function can be extended to validate other fields.
-func (rs *RemoteSecret) ValidateUploadSecretType(uploadSecret *corev1.Secret) error {
-	defaultize := func(secretType corev1.SecretType) corev1.SecretType {
-		if secretType == "" {
-			return corev1.SecretTypeOpaque
+func (rs *RemoteSecret) ValidateUploadSecret(uploadSecret *corev1.Secret) error {
+	if err := checkMatchingSecretTypes(rs.Spec.Secret.Type, uploadSecret.Type); err != nil {
+		return err
+	}
+	if err := rs.ValidateSecretData(uploadSecret.Data); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateSecretData checks weather the secret data contains all the keys required by the secret type and specified
```

Comment:
Either "ValidateSecretData checks weather, the secret data ..." or "ValidatedSecretData checks whether the secret data ...". If it should be the former, I would argue it deserves its own PR 😄

Reply:
The number of times I made this typo might suggest that its time to start coding some weather app. 

Fixed :)